File: backend/app/schemas/transaction.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List, Any, Dict


class TransactionItem(BaseModel):
    name: str
    quantity: int = 1
    price: float
    category: Optional[str] = None


class TransactionCreate(BaseModel):
    id: str
    merchant_id: str
    amount: float
    tip: float = 0.0
    tax: float = 0.0
    currency: str = "USD"
    payment_method: Optional[str] = None
    card_type: Optional[str] = None
    status: str = "completed"
    customer_id: Optional[str] = None
    customer_name: Optional[str] = None
    items: Optional[List[TransactionItem]] = None
    transaction_at: datetime
    clover_order_id: Optional[str] = None
    raw_data: Optional[Dict[str, Any]] = None


class TransactionResponse(BaseModel):
    id: str
    merchant_id: str
    amount: float
    tip: float
    tax: float
    currency: str
    payment_method: Optional[str]
    card_type: Optional[str]
    status: str
    customer_name: Optional[str]
    items: Optional[List[Any]]
    item_count: int
    transaction_at: datetime
    created_at: datetime

    class Config:
        from_attributes = True


class CloverWebhookPayload(BaseModel):
    merchantId: str
    type: str                 # "CREATE", "UPDATE", "DELETE"
    objectId: str             # ID del objeto modificado
    data: Optional[Dict[str, Any]] = None

File: android/app/src/main/java/com/smartreceipt/track/models/Transaction.java
package com.smartreceipt.track.models;

import com.google.gson.annotations.SerializedName;
import java.util.List;

public class Transaction {

    @SerializedName("id")
    private String id;

    @SerializedName("merchant_id")
    private String merchantId;

    @SerializedName("amount")
    private double amount;

    @SerializedName("tip")
    private double tip;

    @SerializedName("tax")
    private double tax;

    @SerializedName("currency")
    private String currency;

    @SerializedName("payment_method")
    private String paymentMethod;

    @SerializedName("card_type")
    private String cardType;

    @SerializedName("status")
    private String status;

    @SerializedName("customer_name")
    private String customerName;

    @SerializedName("items")
    private List<TransactionItem> items;

    @SerializedName("item_count")
    private int itemCount;

    @SerializedName("transaction_at")
    private String transactionAt;

    public String getId()            { return id; }
    public String getMerchantId()    { return merchantId; }
    public double getAmount()        { return amount; }
    public double getTip()           { return tip; }
    public double getTax()           { return tax; }
    public String getCurrency()      { return currency; }
    public String getPaymentMethod() { return paymentMethod; }
    public String getCardType()      { return cardType; }
    public String getStatus()        { return status; }
    public String getCustomerName()  { return customerName; }
    public List<TransactionItem> getItems() { return items; }
    public int getItemCount()        { return itemCount; }
    public String getTransactionAt() { return transactionAt; }

    public static class TransactionItem {
        @SerializedName("name")     private String name;
        @SerializedName("price")    private double price;
        @SerializedName("quantity") private int quantity;
        @SerializedName("category") private String category;

        public String getName()     { return name; }
        public double getPrice()    { return price; }
        public int getQuantity()    { return quantity; }
        public String getCategory() { return category; }
    }
}
